### PR TITLE
Uploaded image preview placeholder

### PR DESCRIPTION
**Pull request by okansahin**
@parasharrajat @JmillsExpensify 

### Details

Showing image preview until image uploaded.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7463

### Tests | QA Steps

1. Open a chat window
2. Click (+ button / add attachment / choose from gallery)
3. Select an image
4. Click Send button in preview modal

- [X] Verify that no errors appear in the JS console

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

##### Chrome

https://user-images.githubusercontent.com/1966034/152587003-b24c3926-cdb7-48b5-a61d-c48fd2156fb1.mov

##### Safari

https://user-images.githubusercontent.com/1966034/152587030-1776db53-05d2-476e-abff-bb166cce103f.mov

#### Mobile Web

https://user-images.githubusercontent.com/1966034/152587895-e1b94cfc-fbd9-4a8e-8596-55984cf06427.mov

#### Desktop

https://user-images.githubusercontent.com/1966034/152587069-d1742004-79a6-4775-8f9c-61fd0d1525e9.mov

#### iOS

https://user-images.githubusercontent.com/1966034/152587121-d8bd1300-ef62-4856-bbca-449f24d56443.mov

#### Android

https://user-images.githubusercontent.com/1966034/152587159-79a23d4d-4a07-48cb-bed8-31e86667e121.mov

### CODE REVIEW

**Comment by okansahin** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by parasharrajat** [comment]
cc: @shawnborton Could you please review the design of the image preview?

**Comment by shawnborton** [comment]
Is there a way we can make this more smooth? Could we detect the image dimensions upon upload and at least make it so that the preview area doesn't jump around as much?

Also, I'm curious about ActivityIndicator we're using here. Do we have other examples of where we use it in the app and it has a solid background behind it? I would think we would just show the loading spinner on top of the content with no box around it. Let me know if that makes sense.

**Comment by okansahin** [comment]
@shawnborton 

> Is there a way we can make this more smooth? Could we detect the image dimensions upon upload and at least make it so that the preview area doesn't jump around as much?

Yes It can, but it is a different issue than this.

> 
> Also, I'm curious about ActivityIndicator we're using here. Do we have other examples of where we use it in the app and it has a solid background behind it? I would think we would just show the loading spinner on top of the content with no box around it. Let me know if that makes sense.

The activity indicator is default by operating system. I added a faded background because it's hard to see with photos which have light colors. (https://github.com/Expensify/App/issues/7463#issuecomment-1027975719)

If you provide me a design, I can set it to it.

**Comment by shawnborton** [comment]
Why would you consider that a different issue than this?

**Comment by parasharrajat** [comment]
Ok. yeah. Thanks for your thoughts @okansahin. 

> Is there a way we can make this more smooth? Could we detect the image dimensions upon upload and at least make it so that the preview area doesn't jump around as much?

I agree that this is not part of this PR and requires a bigger implementation change. But this is a good suggestion we can talk about this more on slack. 

> Also, I'm curious about ActivityIndicator we're using here. Do we have other examples of where we use it in the app and it has a solid background behind it? I would think we would just show the loading spinner on top of the content with no box around it. Let me know if that makes sense.

Let's try to work on this. @okansahin Do you have any suggestions?

**Comment by okansahin** [comment]
> 
> > Also, I'm curious about ActivityIndicator we're using here. Do we have other examples of where we use it in the app and it has a solid background behind it? I would think we would just show the loading spinner on top of the content with no box around it. Let me know if that makes sense.
> 
> Let's try to work on this. @okansahin Do you have any suggestions?

I thought this might be good for the ActivityIndicator to appear with every image clear. 

If you think the same with me that faded background is required I might fill the faded background over image. But as @shawnborton said this usage is not exists with the other ActivityIndicators. It cause degradation of the design integrity.

We can use another spinner that you will choose, maybe with Lottie.

Or I can completely remove the faded background.

I think @shawnborton should decide this.

Thank you

**Comment by shawnborton** [comment]
Okay that's a fair comment that we can follow up about dimensions in another PR, but what I found surprising with this animation is that we go from this shape:
image


Then we get a new shape and a blank preview (which we should definitely avoid):
image

Then we land on a final shape like this:
image


I think we should try to avoid the shape changing as much as we possibly can. Do you have any ideas for that?

For the ActivityIndicator, could we just try it without the background fill behind it?

**Comment by okansahin** [comment]
> Okay that's a fair comment that we can follow up about dimensions in another PR, but what I found surprising with this animation is that we go from this shape: image
> 
> Then we get a new shape and a blank preview (which we should definitely avoid): image
> 
> Then we land on a final shape like this: image
> 
> I think we should try to avoid the shape changing as much as we possibly can. Do you have any ideas for that?

This catches my attention too. Even if we apply the solutions I wrote above, this will continue to happen. It requires intervention to the general structure in chat window. If you open a topic in Slack I will be glad to research and participate in the solution.

Even you don't upload a picture, this issue also happens when you open the app and click a chat room. (Here maybe retrieving images from the cache will fix the problem.)

> 
> For the ActivityIndicator, could we just try it without the background fill behind it?

I will do it in a short time and send the screenshots in here.

**Comment by okansahin** [comment]
### Simulator iOS
Screen Shot 2022-02-07 at 17 18 03

### Desktop Chrome
Screen Shot 2022-02-07 at 17 24 24

**Comment by okansahin** [comment]
> The spinner looks a bit too transparent on iOS, but looks fine in Chrome IMO. 
> 
So, what do you want me to change. Is it okay like this?

> 
> 
> > This catches my attention too. Even if we apply the solutions I wrote above, this will continue to happen. It requires intervention to the general structure in chat window. If you open a topic in Slack I will be glad to research and participate in the solution.
> 
> >
> 
> > Even you don't upload a picture, this issue also happens when you open the app and click a chat room. (Here maybe retrieving images from the cache will fix the problem.)
> 
> 
> 
> Hmm but I think where I am a bit confused is that in my post above, why does the second screenshot happen? Why can't we just go from 1 to 3 and this way the size only changes once? 

Because it's about the way of loading messages. This "way" not relational the code we have changed in this PR.

**Comment by shawnborton** [comment]
Maybe for iOS, we can use our brand dark color for the base spinner color? 

> Because it's about the way of loading messages. This "way" is not relational the code we have changed in this PR.

That's fair, I think we should create a separate follow up issue to address that then.

**Comment by okansahin** [comment]
> Maybe for iOS, we can use our brand dark color for the base spinner color?
> 

It's looks like this with dark color which is defined in project. (#0b1b34) 

Screen Shot 2022-02-08 at 11 09 03

**Comment by shawnborton** [comment]
That definitely looks hard to see. Could we see it using one of our light gray colors?

**Comment by okansahin** [comment]
> That definitely looks hard to see. Could we see it using one of our light gray colors?

It's #FAFAFA which is applied.

Screen Shot 2022-02-08 at 11 24 34

**Comment by shawnborton** [comment]
I think that looks better. What do you think?

**Comment by okansahin** [comment]
Yes, it looks better. 

I push code to repo if you agree with it.

**Comment by shawnborton** [comment]
Yup, I agree.

**Comment by okansahin** [comment]
@parasharrajat @Julesssss 

waiting for you review. thank you.

**Comment by Julesssss** [comment]
The iOS E2E tests failed with a timeout, just restarted them. Code looks pretty good to me other than that.

It would be good to make the follow-up improvements that yourself and Shawn discussed.

**Comment by okansahin** [comment]
> The iOS E2E tests failed with a timeout, just restarted them. Code looks pretty good to me other than that.
> 

I saw that dotenv module not found in logs. Do I should merge main branch to this?

> It would be good to make the follow-up improvements that yourself and Shawn discussed.

If you talking about for image jumping, I can create new bug request with that issue if you accept.

**Comment by Julesssss** [comment]
> Do I should merge main branch to this?

I didn't look into the errors, but if you think it will help then please do this.

**Comment by okansahin** [comment]
> I didn't look into the errors, but if you think it will help then please do this.

I didn't merged, all checks have passed now.

thank you @Julesssss

File: src/pages/home/report/ReportActionItemFragment.js
Comment (parasharrajat) [inline src/pages/home/report/ReportActionItemFragment.js:34]: Please add comments to properties.

File: src/pages/home/report/ReportActionItemFragment.js
Comment (parasharrajat) [inline src/pages/home/report/ReportActionItemFragment.js:17]: use variables instead of colors.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

File: src/pages/home/report/ReportActionItemFragment.js
Comment (okansahin) [reply src/pages/home/report/ReportActionItemFragment.js:17]: I double checked it if colors.js are used in components. And I saw in so many files with this usage.

Do you want me to add a new variable as a theme variable?

**Comment by parasharrajat** [comment]
Testing shortly...

**Review by parasharrajat — COMMENTED** [review]
image
Does this look fine @shawnborton? Or should we add a faded background over the image?

**Comment by shawnborton** [comment]
I think that's probably fine. In a separate follow up issue, it would be nice to standardize on the same style spinner everywhere instead of having differences between platforms.

**Review by parasharrajat — APPROVED** [review]
LGTM. 

cc: @Julesssss 
:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by Julesssss — APPROVED** [review]
LGTM

**Pull request merged**
